### PR TITLE
 eliminated duplicate records

### PR DESCRIPTION
**Pull request by tarunsamanta2k20**
## Check List (Check all the applicable boxes) 

- [ ] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] This PR does not contain plagiarized content.
- [ ] The title of my pull request is a short description of the requested changes.

## Note to reviewers

#55

### CODE REVIEW

File: backend/routes/giphy.js [hunk 1]
```diff
@@ -11,8 +10,8 @@ router.route("/").get(async (req, res) => {
       src: img?.images?.downsized?.url || "",
       placeholderSrc: img?.images?.fixed_height_small_still?.url || ""
     }));
-
-    res.json(dataWithSrc);
+    const filterArr = Utils(dataWithSrc);
+    res.json(filterArr);
```
Comment (furkanmutlu) [inline backend/routes/giphy.js:14]: I think it would be better to do filtering on frontend side. Because frontend is requesting data from backend with 'LIMIT'. According to this limit frontend creates Grid. So filtering elements on backend might/will send frontend misleading amount of images. 

What I would recommend is to use `Array.filter` on `Gallery.jsx` line 44.

What do you think?

**Review by furkanmutlu — CHANGES_REQUESTED** [review]

**Comment by tarunsamanta2k20** [comment]
I think no. Because we eliminated duplicate on backend side. So duplicate data will never go to frontend.

File: backend/routes/giphy.js (hunk 1 above)
Comment (tarunsamanta2k20) [reply backend/routes/giphy.js:14]: I think no. Because we eliminated duplicate on backend side. So duplicate data will never go to frontend.

Comment (tarunsamanta2k20) [reply backend/routes/giphy.js:14]: Okay will do changes on frontend.

**Comment by furkanmutlu** [comment]
> I think no. Because we eliminated duplicate on backend side. So duplicate data will never go to frontend.

Yes but thats not my point. My point is our backend should just delivers what it gets, so it will be kind of a proxy, and frontend will receive exact amount of items it requested?

**Comment by tarunsamanta2k20** [comment]
run now

File: frontend/src/components/Gallery/Gallery.jsx [hunk 2]
```diff
@@ -33,9 +33,13 @@ const Gallery = ({ images, apiType }) => {
 
     setFavImages(favImages.filter(favImg => favImg.id !== id));
   };
-
   const hasLike = (id) => favImages && favImages.find(favImg => favImg.id === id);
-
+  const seen = new Set();
+  images=images.filter((el) => {
```
Comment (furkanmutlu) [inline frontend/src/components/Gallery/Gallery.jsx:38]: `images` is a react state here. You should never assign a value to the state directly. I will create a clean PR and mention you to show my solution. For now, you can either make this PR draft or close it.

File: backend/routes/giphy.js [hunk 3]
```diff
@@ -22,13 +20,17 @@ router.route("/").get(async (req, res) => {
 router.route("/search").get(async (req, res) => {
   try {
     const { searchTerm, limit, offset } = req.query;
-    const { data } = await getSearchedImages(apiSources.giphy.id, limit, searchTerm, offset);
+    const { data } = await getSearchedImages(
```
Comment (furkanmutlu) [inline backend/routes/giphy.js:23]: this change is not related to your story. Please just push what is necessary to fix the issue. It makes harder/longer for reviewers to read.

**Review by furkanmutlu — CHANGES_REQUESTED** [review]
For now, you can either make this PR draft or close it.
